### PR TITLE
[14.0][fix] stock: stock.picking.state to draft associating existing picking to a computed field

### PR DESCRIPTION
**Pull request by fredzamoabg**
Description of the issue/feature this PR addresses:
Associating an existing picking with `state` different to `draft` to a computed field (for example a field in `wizard`) the `state` field is recalculated setting the value to `draft`.

The problem is found here https://github.com/odoo/odoo/blob/a05f4e05b9f4df88948ac3dfc1522828b63bfb64/addons/stock/models/stock_picking.py#L485 where is created a structure with the ID picking taken from `stock.move` as key (that is always an `Integer`) but here https://github.com/odoo/odoo/blob/a05f4e05b9f4df88948ac3dfc1522828b63bfb64/addons/stock/models/stock_picking.py#L492 the data is retrieved from structure using ID of `self` that can be of type `New_id`.

Current behavior before PR:
The picking state is incorrect as is always set to `draft`.

Desired behavior after PR is merged:
The picking state remain as expected.
This is obtained adding to the structure the same type of ID used to retrieve it.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by amoyaux** [comment]
Ok but build a dict with link Id -> VirtualId on top of the loop and get the correct Id from the dict instead of doing a filter on each loop (it will slow a lot the function if there is an important amount of pickings)

**Comment by fredzamoabg** [comment]
> Ok but build a dict with link Id -> VirtualId on top of the loop and get the correct Id from the dict instead of doing a filter on each loop (it will slow a lot the function if there is an important amount of pickings)

correct. another solution is always to get from structure passing **Integer**. just a couple o rows in plus but better in terms of performance

**Comment by fredzamoabg** [comment]
@amoyaux can you check new proposal?

File: addons/stock/models/stock_picking.py [hunk 1]
```diff
@@ -489,16 +489,17 @@ def _compute_state(self):
             })
             picking_move_lines[picking_id.id].add(move.id)
         for picking in self:
-            if not picking_moves_state_map[picking.id]:
+            picking_id = (picking.ids and picking.ids[0]) or picking.id
```
Comment (amoyaux) [inline addons/stock/models/stock_picking.py:492]: picking.ids[:1] is not enough?

Comment (fredzamoabg) [reply addons/stock/models/stock_picking.py:492]: `[:1]` returns a list so what do you mean with `enough` ?

Comment (amoyaux) [reply addons/stock/models/stock_picking.py:492]: Oups my bad

**Comment by amoyaux** [comment]
Okay just a last thing, could you rename the commit message to [FIX] stock: stock.picking.state to draft associating existing …
To respect the guidelines

**Comment by fredzamoabg** [comment]
> Okay just a last thing, could you rename the commit message to [FIX] stock: stock.picking.state to draft associating existing … To respect the guidelines

done.

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
@fredzamoabg Thanks for the contribution

**Comment by amoyaux** [comment]
robodoo rebase-ff

**Pull request closed without merging**